json_serializable/lib/src/json_serializable_generator.dart

```diff
@@ -558,3 +583,28 @@ InvalidGenerationSourceError _createInvalidGenerationError(
   return new InvalidGenerationSourceError(message,
       todo: 'Make sure all of the types are serializable.');
 }
+
+/// Returns a list of all instance, [FieldElement] items for [element] and
+/// super classes.
+List<FieldElement> _getFields(ClassElement element) {
+  // Get all of the fields that need to be assigned
+  // TODO: support overriding the field set with an annotation option
+  var fieldsList = element.fields.where((e) => !e.isStatic).toList();
+
+  var manager = new InheritanceManager(element.library);
+  var things = manager.getMembersInheritedFromClasses(element);
+
+  things.forEach((k, v) {
```

Comment:
you don't appear to use `k` at all and this loop has side effects. a for-in loop would be more idiomatic

```
for(final member in things.values) {
  assert(!member.isStatic)
  ...
}
```

Reply:
ack